app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,12 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.new(comment_params)
+    redirect_to user_post_path(id: @comment.post_id, user_id: @comment.author_id) if @comment.save
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text, :author_id, :post_id)
+  end
+end
```

Comment:
Please note that I should have made this comment on the view but I found non so I have to make it on the controller that should be responsible for rendering that view.

- [ ] There should be a form which should accept inputs for creating a comment. That form should be inside the `views/comments/new.html`.  Please create that view.

Please don't forget to define the `new` method in this controller and also a route that will point to it.

see example:
```ruby
def new
  @comment = Comment.new
end
```

example route:
```ruby
/posts/:post_id/comments/new(.:format) comments#new
```

Reply:
Hi @mmsesay,

But, I implemented this functionality as an inline form. The reason why I did not use an independent view is that - by design architecture, we do not need to redirect a user to a new view just to make a comment as we did for creating a new post.

Reply:
Hi @blessedjasonmwanza, I promised to get back after hearing back from a Senior CR.
You can resubmit without making the change since the inline implementation is working as expected.

Reply:
Hi @mmsesay,

Thank you so much. On to doing so />